### PR TITLE
Updating `DataFrame.mode` docstring.

### PR DESCRIPTION
**Pull request by datapythonista**
Supersedes #20241 (source branch does not exist anymore).

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -7213,38 +7213,88 @@ def _get_agg_axis(self, axis_num):
...
+        Notes
+        -----
+        Every column or row of the resulting DataFrame contains all its modes.
+        And possibly NaN values at the end (if other columns or rows have a
+        higher number of modes).
 
         Examples
         --------
-        >>> df = pd.DataFrame({'A': [1, 2, 1, 2, 1, 2, 3]})
+
```
Comment (jbrockmendel) [inline pandas/core/frame.py]: Is it standard to have a newline after the section title?

Comment (datapythonista) [reply pandas/core/frame.py]: No, it's not, good catch. I added it to the TODO list for the validation script in #20298 (it happens very often).

File: pandas/core/frame.py [hunk 2]
```diff
@@ -7213,38 +7213,87 @@ def _get_agg_axis(self, axis_num):
...
         ----------
         axis : {0 or 'index', 1 or 'columns'}, default 0
-            * 0 or 'index' : get mode of each column
-            * 1 or 'columns' : get mode of each row
-        numeric_only : boolean, default False
-            if True, only apply to numeric columns
-        dropna : boolean, default True
+            The axis to iterate over while searching for the mode.
+            To find the mode for each column, use ``axis='index'``.
+            To find the mode for each row, use ``axis='columns'``.
```
Comment (gfyoung) [inline pandas/core/frame.py]: I like the first line of your summary.  However, I would put back the two bullet points.

Comment (datapythonista) [reply pandas/core/frame.py]: Agree, fixed. And we should probably define a standard and be consistent when documenting `axis` (added it to #20298)

File: pandas/core/frame.py [hunk 3]
```diff
@@ -7217,38 +7217,88 @@ def _get_agg_axis(self, axis_num):
...
+        ...                   index=('falcon', 'horse', 'spider', 'ostrich'),
+        ...                   columns=('species', 'legs', 'wings'))
+        >>> df
+                   species  legs  wings
+        falcon        bird     2    2.0
+        horse       mammal     4    NaN
+        spider   arthropod     8    0.0
+        ostrich       bird     2    NaN
+
+        By default, missing values are not considered, and the mode of winds
```
Comment (WillAyd) [inline pandas/core/frame.py]: Think you mean to say `wings` here and not `winds`; side note - ostriches do still have wings :-)

File: pandas/core/frame.py [hunk 4]
```diff
@@ -7217,38 +7217,88 @@ def _get_agg_axis(self, axis_num):
...
+        are both 0 and 2. The second row of species and legs contains NaN,
```
Comment (WillAyd) [inline pandas/core/frame.py]: Double backticks for `NaN`?

File: pandas/core/frame.py [hunk 5]
```diff
@@ -7217,38 +7217,88 @@ def _get_agg_axis(self, axis_num):
...
+            The modes of each column or row.
+
+        See Also
+        --------
+        Series.mode : Return the highest frequency value in a Series.
+        Series.value_counts : Return the counts of values in a Series.
+
+        Notes
+        -----
+        Every column or row of the resulting DataFrame contains all its modes.
```
Comment (WillAyd) [inline pandas/core/frame.py]: May just be me but I don't understand the Notes section - is this necessary or better explained via examples?

File: pandas/core/frame.py [hunk 6]
```diff
@@ -7217,38 +7217,88 @@ def _get_agg_axis(self, axis_num):
...
-        for each mode per label, fills in gaps with nan.
+        Get the mode(s) of each element along the selected axis.
 
-        Note that there could be multiple values returned for the selected
-        axis (when more than one item share the maximum frequency), which is
-        the reason why a dataframe is returned. If you want to impute missing
-        values with the mode in a dataframe ``df``, you can just do this:
-        ``df.fillna(df.mode().iloc[0])``
+        The mode of a set of values is the value that appears most often.
+        It can be multiple values.
```
Comment (WillAyd) [inline pandas/core/frame.py]: Can we add an example to show this?

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Review by datapythonista — COMMENTED** [review]
addressed changes from code review.

**Comment by datapythonista** [comment]
@WillAyd can you take a look and merged this, if the comments from your review are addressed correctly. Thanks!

**Pull request merged**